demo/style_configurator.c

```diff
@@ -0,0 +1,1048 @@
+	style_vec2(ctx, "Touch Padding:", &select.touch_padding);
+
+	nk_property_float(ctx, "#Rounding:", -100.0f, &select.rounding, 100.0f, 1,0.5f);
+
+
+	*out_style = select;
+}
+
+static void
+style_slider(struct nk_context* ctx, struct nk_style_slider* out_style)
+{
+	struct nk_style_slider slider = *out_style;
+	//slider = &style->slider;
+	//slider->normal          = nk_style_item_hide();
+	//slider->hover           = nk_style_item_hide();
+	//slider->active          = nk_style_item_hide();
+	//slider->bar_normal      = table[NK_COLOR_SLIDER];
+	//slider->bar_hover       = table[NK_COLOR_SLIDER];
+	//slider->bar_active      = table[NK_COLOR_SLIDER];
+	//slider->bar_filled      = table[NK_COLOR_SLIDER_CURSOR];
+	//slider->cursor_normal   = nk_style_item_color(table[NK_COLOR_SLIDER_CURSOR]);
+	//slider->cursor_hover    = nk_style_item_color(table[NK_COLOR_SLIDER_CURSOR_HOVER]);
+	//slider->cursor_active   = nk_style_item_color(table[NK_COLOR_SLIDER_CURSOR_ACTIVE]);
+	//slider->inc_symbol      = NK_SYMBOL_TRIANGLE_RIGHT;
+	//slider->dec_symbol      = NK_SYMBOL_TRIANGLE_LEFT;
+	//slider->cursor_size     = nk_vec2(16,16);
+	//slider->padding         = nk_vec2(2,2);
+	//slider->spacing         = nk_vec2(2,2);
+	//slider->userdata        = nk_handle_ptr(0);
+	//slider->show_buttons    = nk_false;
+	//slider->bar_height      = 8;
+	//slider->rounding        = 0;
+	//slider->draw_begin      = 0;
+	//slider->draw_end        = 0;
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+
+	style_item_color(ctx, "Normal:", &slider.normal);
+	style_item_color(ctx, "Hover:", &slider.hover);
+	style_item_color(ctx, "Active:", &slider.active);
+
+	style_rgb(ctx, "Bar Normal:", &slider.bar_normal);
+	style_rgb(ctx, "Bar Hover:", &slider.bar_hover);
+	style_rgb(ctx, "Bar Active:", &slider.bar_active);
+	style_rgb(ctx, "Bar Filled:", &slider.bar_filled);
+
+	style_item_color(ctx, "Cursor Normal:", &slider.cursor_normal);
+	style_item_color(ctx, "Cursor Hover:", &slider.cursor_hover);
+	style_item_color(ctx, "Cursor Active:", &slider.cursor_active);
+
+	style_vec2(ctx, "Cursor Size:", &slider.cursor_size);
+	style_vec2(ctx, "Padding:", &slider.padding);
+	style_vec2(ctx, "Spacing:", &slider.spacing);
+
+	nk_property_float(ctx, "#Bar Height:", -100.0f, &slider.bar_height, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Rounding:", -100.0f, &slider.rounding, 100.0f, 1,0.5f);
+
+	nk_layout_row_dynamic(ctx, 30, 1);
+	nk_checkbox_label(ctx, "Show Buttons", &slider.show_buttons);
+
+	if (slider.show_buttons) {
+		nk_layout_row_dynamic(ctx, 30, 2);
+		nk_label(ctx, "Inc Symbol:", NK_TEXT_LEFT);
+		slider.inc_symbol = nk_combo(ctx, symbols, NK_SYMBOL_MAX, slider.inc_symbol, 25, nk_vec2(200,200));
+		nk_label(ctx, "Dec Symbol:", NK_TEXT_LEFT);
+		slider.dec_symbol = nk_combo(ctx, symbols, NK_SYMBOL_MAX, slider.dec_symbol, 25, nk_vec2(200,200));
+
+		// necessary or do tree's always take the whole width?
+		//nk_layout_row_dynamic(ctx, 30, 1);
```

Comment:
Hmm. Good question. If I add a label that I want the tree to share width with the tree get a new row.
![image](https://user-images.githubusercontent.com/3759175/76239061-290f2780-6231-11ea-8ccd-f128d5b44ed7.png)

Now, this might be intended but I don't know.